src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+    if ((Message_GetState(&play->msgCtx) == 5) && Message_ShouldAdvance(play)) {
+        if ((this->actionFlags & ENSGORO_ACTIONFLAG_LASTMESSAGE)) {
+            this->actionFlags &= ~ENSGORO_ACTIONFLAG_LASTMESSAGE;
+            this->actionFlags &= ~ENSGORO_ACTIONFLAG_ENGAGED;
+            play->msgCtx.msgMode = 0x43;
+            play->msgCtx.unk12023 = 4;
+            this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+            return;
+        }
+        this->textID = EnSGoro_ShrineGoron_NextTextID(this, play);
+        Message_StartTextbox(play, this->textID, &this->actor);
+    }
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void EnSGoro_ActionShopGoron(EnSGoro* this, PlayState* play) {
+    if (!(this->actionFlags & ENSGORO_ACTIONFLAG_ROLLEDUP)) {
+        EnSGoro_UpdateToIdleAnimation(this);
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFlags |= ENSGORO_ACTIONFLAG_ENGAGED;
+        this->textID = EnSGoro_BombshopGoron_NextTextID(this, play);
+        if ((this->actionFlags & ENSGORO_ACTIONFLAG_ROLLEDUP)) {
+            this->actionFlags &= ~ENSGORO_ACTIONFLAG_ROLLEDUP;
+            this->animInfoIndex = ENSGORO_ANIM_UNROLL_A;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+            this->actionFunc = EnSGoro_ActionShopGoronFinishUnroll;
+        } else {
+            Message_StartTextbox(play, this->textID, &this->actor);
+            this->actionFunc = EnSGoro_ActionShopGoronTalking;
+        }
+    } else if ((this->actor.xzDistToPlayer < 250.0f) || this->actor.isTargeted) {
+        func_800B863C(&this->actor, play);
+    }
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void EnSGoro_ActionShopGoronFinishUnroll(EnSGoro* this, PlayState* play) {
+
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if ((this->animInfoIndex == ENSGORO_ANIM_UNROLL_A) && (curFrame == lastFrame)) {
+        this->animInfoIndex = ENSGORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+        Message_StartTextbox(play, this->textID, &this->actor);
+        this->actionFunc = EnSGoro_ActionShopGoronTalking;
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void EnSGoro_ActionShopGoronTalking(EnSGoro* this, PlayState* play) {
+
+    u8 msgState = Message_GetState(&play->msgCtx);
+
+    if ((this->actionFlags & ENSGORO_ACTIONFLAG_HANDTAP)) {
+        EnSGoro_UpdateToHandtapAnimation(this);
+    } else if ((this->actionFlags & ENSGORO_ACTIONFLAG_TIRED)) {
```

Comment:
```suggestion
    if (this->actionFlags & ENSGORO_ACTIONFLAG_HANDTAP) {
        EnSGoro_UpdateToHandtapAnimation(this);
    } else if (this->actionFlags & ENSGORO_ACTIONFLAG_TIRED) {
```
I'm finding quite a few of these extra unnecessary parenthesis, would you mind double checking else where as well.